builder/parser/line_parsers.go

```diff
@@ -195,6 +194,35 @@ func parseLabel(rest string) (*Node, map[string]bool, error) {
 	return parseNameVal(rest, "LABEL")
 }
 
+// parses a statement with multiple keywords definition and name-value
+// assignment, like `name1 name2 name3=value`.
+// Note that this is a stricter format than the old format of assignment, in a way
+// that it only allows assignment of the form `name=value`. And that a name
+// definition or assignment to an empty value is designated by `name`.
```

Comment:
@mapuri okay it makes more sense now. Both https://github.com/docker/docker/pull/15182#discussion_r39445160 or the revised form is fine for me, but I'd like to name the function that uses the "old format assignment".